docs/current/guides/281969-dagger-on-kubernetes.md

```diff
@@ -0,0 +1,218 @@
+      securityContext:
+        runAsUser: 0
+        runAsGroup: 1001
+        fsGroup: 1001
+        fsGroupChangePolicy: "OnRootMismatch"
+      initContainers:
+        - name: download-bin
+          image: {{ .Values.dagger.engine.image.repository }}:{{ .Values.dagger.engine.image.tag }}
+          command: ["sh", "-c", "wget https://github.com/moby/buildkit/releases/download/v0.11.6/buildkit-v0.11.6.linux-amd64.tar.gz -O /shared-bin/buildkit.tar.gz && tar -xvzf /shared-bin/buildkit.tar.gz --strip-components 1 bin/buildctl -C /shared-bin"]
+          volumeMounts:
+          - name: shared-bin
+            mountPath: /shared-bin
+      containers:
+        - name: dagger-engine
+          image: {{ .Values.dagger.engine.image.repository }}:{{ .Values.dagger.engine.image.tag }}
+          imagePullPolicy: Always
+          args:
+            - "--oci-max-parallelism"
+            - "num-cpu"
+          {{- if and .Values.dagger.magicache.enabled .Values.dagger.magicache.cloud}}
+          env:
+          - name: _EXPERIMENTAL_DAGGER_CACHESERVICE_URL
+            value: {{ required "A magicache url is required" .Values.dagger.magicache.cloud.url }}
+          - name: _EXPERIMENTAL_DAGGER_CACHESERVICE_TOKEN
+            value: {{ required "A magicache token is required" .Values.dagger.magicache.cloud.token }}
+          {{- end }}
+          securityContext:
+            privileged: true
+            capabilities:
+              add:
+                - ALL
+          readinessProbe:
+            exec:
+              command: ["/shared-bin/buildctl", "debug", "workers"]
+            initialDelaySeconds: 5
+            periodSeconds: 5
+          volumeMounts:
+            - name: varlibdagger
+              mountPath: /var/lib/dagger
+            - name: varrundagger
+              mountPath: /var/run/buildkit
+            - name: shared-bin
+              mountPath: /shared-bin
+      terminationGracePeriodSeconds: 300
+      volumes:
+        - name: shared-bin
+          emptyDir: {}
+        - name: varlibdagger
+          hostPath:
+            path: /var/lib/dagger
+        - name: varrundagger
+          hostPath:
+            path: /var/run/dagger
+```
+
+### Taints and Tolerations
+
+We use these to ensure that:
+
+- Runners are only scheduled on runner nodes
+- Dagger Engines are co-located with runners
+
+We're using a DaemonSet for the Dagger Engine to
+- best utilize the NVMe hard drives of our worker nodes
+- reduce the amount of network latency and bandwidth requirements
+- simplify routing of Dagger SDK and CLI requests
+
+We use the `_EXPERIMENTAL_DAGGER_RUNNER_HOST` environment variable to point to the Dagger Engine DaemonSet socket that we've mounted into our GitHub Runners. This ensures that our runners will use the local Dagger Engine for pipeline execution.
+
+## Design Decisions
+
+- We've chosen to use a Dagger Cloud Cache, which allows us to push cached layers up to the cloud and retrieve them at high bandwidth.
```

Comment:
What if the user doesn't have Dagger Cloud? Is there an alternative caching mechanism?